Framework/DataHandling/src/ISISDataArchive.cpp

```diff
@@ -91,7 +97,7 @@ std::ostringstream ISISDataArchive::sendRequest(const std::string &fName) const
   Kernel::InternetHelper inetHelper;
   std::ostringstream os;
   try {
-    inetHelper.sendRequest(URL_PREFIX + fName, os);
+    inetHelper.sendRequest(std::string(URL_PREFIX) + fName, os);
```

Comment:
Out of interest, do you know if this creates an additional `std::string` of `URL_PREFIX` or just it optimise away and just create one of `URL_PREFIX + fName`?

Reply:
I'm not sure, actually. I know it can't append anything to the `string_view` object, so I think it probably makes the additional string and then appends them? I tried to have a look in the [compiler explorer](https://godbolt.org/) but I'm not overly familiar with assembly to know what it's doing

Reply:
Looks like it does:
```
        call    std::allocator<char>::allocator() [complete object constructor]
        mov     rdx, qword ptr [rbp - 240]      # 8-byte Reload
        lea     rdi, [rbp - 208]
        lea     rsi, [rbp - 144]
        call    std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string<std::basic_string_view<char, std::char_traits<char> >, void>(std::basic_string_view<char, std::char_traits<char> > const&, std::allocator<char> const&)
        jmp     .LBB2_5
.LBB2_5:
        lea     rdi, [rbp - 176]
        lea     rsi, [rbp - 208]
        lea     rdx, [rbp - 88]
        call    std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
        jmp     .LBB2_6
.LBB2_6:
        lea     rdi, [rbp - 176]
        call    str_function(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
        jmp     .LBB2_7
.LBB2_7:
        lea     rdi, [rbp - 176]
        call    std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string() [base object destructor]
        lea     rdi, [rbp - 208]
        call    std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string() [base object destructor]
        lea     rdi, [rbp - 216]
        call    std::allocator<char>::~allocator() [complete object destructor]
```

vs

```
        call    std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
        jmp     .LBB2_5
.LBB2_5:
        lea     rdi, [rbp - 176]
        call    str_function(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
        jmp     .LBB2_6
.LBB2_6:
        lea     rdi, [rbp - 176]
        call    std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string() [base object destructor]
```

Reply:
Turns out the best way to do this is `inetHelper.sendRequest(URL_PREFIX.data() + fName, os)`.

Can we do that?